internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -962,3 +973,64 @@ func (md *machineDeployment) doSmokeChecks(ctx context.Context, lm machine.Leasa
 
 	return fmt.Errorf("smoke checks for %s failed: %v", lm.Machine().ID, err)
 }
+
+func (md *machineDeployment) checkDNS(ctx context.Context) error {
+	client := client.FromContext(ctx).API()
+	ipAddrs, err := client.GetIPAddresses(ctx, md.appConfig.AppName)
+	if err != nil {
+		return err
+	}
+
+	if url := md.appConfig.URL(); url != nil && len(ipAddrs) > 0 {
```

Comment:
I typically try to avoid shadowing stdlib package names (e.g. `url`). A better name might be `u` (if you're feeling terse) or `appURL`.